CMakeLists.txt

```diff
@@ -1572,7 +1572,7 @@ endif()
 target_include_directories(mixxx-lib PUBLIC lib/rigtorp/SPSCQueue/include)
 
 # Qt
-find_package(Qt5
+find_package(Qt5 5.12.0
```

Comment:
Is there a workaround for QT 5.9.5 and Ubuntu bionic? 
Instead of require a specific Qt version in the make file we should make use of QT_VERSION for conditional builds.

Reply:
No, Qt 5.12 is now a strict requirement with this branch. There is no way to hack around that without disabling the whole feature, which I will not do. In #2682, I added a quick hack to keep master building with Qt < 5.12 because the new Qt 5.12 feature was not essential and the Ubuntu 20.04 upgrade was not yet available. But now the Ubuntu 20.04 upgrade is available, so I will not maintain support for Qt < 5.12 in master.

Reply:
Which feature exactly is required and why?

Reply:
Everything. This entire PR is useless without the QJSEngine::importModule function introduced in Qt 5.12.

Reply:
Why can't we import the files a scripts as usual? What is the benefit of modules in our case?

Reply:
The `import` statement in JS only works if we use QJSEngine::importModule. The legacy method of reading a file to a QString and passing it to QJSEngine::evaluate does not support JS `import`.

Reply:
Isn't the import detective a ES6 feature? 
So I think If we write a controller script that makes use of import, it requires a Mixxx version based on QT 5.12. 
All other scripts can still run with older QT Versions. 

But what is exactly the benefit of modules in the Mixxx case? 
It is probably less confusing, if everything is loaded by default.

Reply:
> Isn't the import detective a ES6 feature?

Yes. But in my testing it only worked with QJSEngine::importModule, not QJSEngine::evaluate.

> But what is exactly the benefit of modules in the Mixxx case?

Organizing code, just like any other JS context.

Reply:
These discussions have been going on for months so it is a bit frustrating to have the premise of this PR questioned at this point.

Reply:
Sorry, I was not aware of that. At least the commit that forces Qt 5.12 is from yesterday.  

> Organizing code, just like any other JS context.

IMHO we can live without that. For my feeling it would be even easier for the script authors if everything is just there, instead of dealing with different modules.

Reply:
I'm not going to debate whether we should use JS modules.

Reply:
IMHO the current system is an atrocious hack and JS modules are they way to go. That said, if  @daschuer has a different opinion we should discuss on Zulip and reason why this chance is needed, so that we can try to reach a concensus. @daschuer if you wish to discuss this, please post on Zulip.

Reply:
Why is this only being questioned now?? There have been numerous prior discussions about this.

Reply:
We have already agreed before that the minimum system requirements for Mixxx releases will be those of the latest Ubuntu LTS unless the upgrade to the new LTS is not available yet. The upgrade to Ubuntu 20.04 is now available, therefore it is the minimum for Mixxx 2.4 and 5.12 is the minimum Qt version.

Reply:
This has two aspects. 
First I am in doubt that a controller needs to import modules. Which modules do you like to import? 
If it turns out that there is a strong need for it, there is no strong need to drop drop Ubuntu Bionic support for it. 
We can guard it by a QT_VERSION macro. 
For my feeling there will only be a minority or mappings that load additional modules. So it would be fine to ask users to update their system, if they own such a controller.

Reply:
All mappings will be modules and so will the JS libraries we ship with Mixxx.

Reply:
> If it turns out that there is a strong need for it, there is no strong need to drop drop Ubuntu Bionic support for it.
We can guard it by a QT_VERSION macro.

No. I will not maintain hacks for systems older than our established minimum requirements policy.

> For my feeling there will only be a minority or mappings that load additional modules. So it would be fine to ask users to update their system, if they own such a controller.

All new mappings will be required to be modules when the new system is implemented.

Reply:
> We have already agreed before that the minimum system requirements for Mixxx releases will be those of the latest Ubuntu LTS unless the upgrade to the new LTS is not available yet. The upgrade to Ubuntu 20.04 is now available, therefore it is the minimum for Mixxx 2.4 and 5.12 is the minimum Qt version.

No, sorry.

Reply:
I agree, but maybe @daschuer wasn't involved in these discussions and I'd rather discuss this on Zulip than in a GitHub review comment.  Don't worry, I'm sure we can figure this out. If missed a discussion, I'd expect the rest of the team to hear me out and explain their reasoning, too. If we can't reach a concensus, we'll vote.

@daschuer If you have objections about this PR, please open a topic on Zulip so that we discuss the problem and hopefully figure it out. OK?

Reply:
@daschuer was involved in that discussion https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/minimum.20requirements.20policy

Reply:
Okay, I reread that old discussion and apparently we didn't reach consensus on a policy. This is not acceptable. We need a policy. It is extremely frustrating to discuss the same topic years later.

Reply:
@daschuer do you have objection against JS modules other than increasing the minimum Qt version? If so, please open a topic on Zulip.